pkgs/development/compilers/riscv-gnu-toolchain/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+
+, gcc12
+, glibc
+, newlib
+, gdb
+, qemu
+, musl
+, spike
+, riscv-pk
+
+, autoconf
+, curl
+, automake
+, python3
+, libmpc
+, gawk
+, bison
+, flex
+, texinfo
+, gperf
+, libtool
+, expat
+, gmp
+, mpfr
+, flock
+, zlib
+
+, srcs ? {
+  binutils = {
+    url = "mirror://gnu/binutils/binutils-2.39.tar.bz2";
+    hash = "sha256-2iSoT+8iAQLdJAQt8G/eqFHCYUpTd/hu/6KPM7exYUg=";
+  };
+  dejagnu = {
+    url = "mirror://gnu/dejagnu/dejagnu-1.6.3.tar.gz";
+    hash = "sha256-h9rvrNeVi0pp+IxoVtvRY0JhljxBQHnQw3H1ic1mouM=";
+  };
+  gcc = gcc12.cc.src;
+  glibc = glibc.src;
+  newlib = newlib.src;
+  gdb = gdb.src;
+  qemu = qemu.src;
+  musl = musl.src;
+  spike = spike.src;
+  pk = riscv-pk.src;
+}
```

Comment:
Please move this out of the inputs. It is not really extensible especially when you want to change one variable.

Reply:
could you not use `pkg.override (old: new: {srcs=old.srcs//{new-pkg=new-pkg-src}})`?

Reply:
This is a bit hacky. Normally src and srcs are exclusive and I would rather expect that overwriting gcc and its source would affect it. Generally it is not such a great idea to put big defaults behind ``?``.

Reply:
Is there a better mechanism, or is the ability to change the sources uneeded?